evm/src/test/java/org/hyperledger/besu/evm/operations/SelfDestructOperationTest.java

```diff
@@ -0,0 +1,185 @@
+
+    when(account.getMutable()).thenReturn(mutableAccount);
+    Address contractAddress = Address.fromHexString(contract);
+    Address beneficiaryAddress = Address.fromHexString(beneficiary);
+    messageFrame =
+        MessageFrame.builder()
+            .type(MessageFrame.Type.CONTRACT_CREATION)
+            .contract(Address.ZERO)
+            .inputData(Bytes.EMPTY)
+            .sender(beneficiaryAddress)
+            .value(Wei.ZERO)
+            .apparentValue(Wei.ZERO)
+            .code(CodeFactory.createCode(SELFDESTRUCT_CODE, 0, true))
+            .depth(1)
+            .completer(__ -> {})
+            .address(contractAddress)
+            .blockHashLookup(n -> Hash.hash(Words.longBytes(n)))
+            .blockValues(mock(BlockValues.class))
+            .gasPrice(Wei.ZERO)
+            .messageFrameStack(new ArrayDeque<>())
+            .miningBeneficiary(Address.ZERO)
+            .originator(Address.ZERO)
+            .initialGas(100_000L)
+            .worldUpdater(worldUpdater)
+            .build();
+    messageFrame.pushStackItem(Bytes.fromHexString(beneficiary));
+
+    when(mutableAccount.getBalance()).thenReturn(Wei.fromHexString(balanceHex));
+    when(mutableAccount.getAddress()).thenReturn(contractAddress, beneficiaryAddress);
+    when(worldUpdater.getAccount(any())).thenReturn(account);
+    when(worldUpdater.get(any())).thenReturn(account);
+    when(worldUpdater.getOrCreate(any())).thenReturn(account);
+  }
+
+  public static Object[][] params() {
+    return new Object[][] {
+      {
+        "0x00112233445566778899aabbccddeeff11223344",
+        "0x1234567890abcdef1234567890abcdef12345678",
+        true,
+        "0x1234567890"
+      },
+      {
+        "0x00112233445566778899aabbccddeeff11223344",
+        "0x1234567890abcdef1234567890abcdef12345678",
+        false,
+        "0x1234567890"
+      },
+      {
+        "0x00112233445566778899aabbccddeeff11223344",
+        "0x00112233445566778899aabbccddeeff11223344",
+        true,
+        "0x1234567890"
+      },
+      {
+        "0x1234567890abcdef1234567890abcdef12345678",
+        "0x1234567890abcdef1234567890abcdef12345678",
+        false,
+        "0x1234567890"
+      },
+    };
+  }
+
+  @ParameterizedTest
+  @MethodSource("params")
+  void checkContractDeletionFrontier(
+      final String contract,
+      final String beneficiary,
+      final boolean ignoredNewAccount,
+      final String balanceHex) {
+
+    setUp(contract, beneficiary, balanceHex);
+    final Operation.OperationResult operationResult = operation.execute(messageFrame, evm);
+    assertThat(operationResult).isNotNull();
+    assertThat(messageFrame.getSelfDestructs()).contains(Address.fromHexString(contract));
+
+    verify(account).getBalance();
+    verify(account).isEmpty();
+    if (!contract.equals(beneficiary)) {
+      verify(mutableAccount).incrementBalance(Wei.fromHexString(balanceHex));
```

Comment:
The variables `account` and  `mutableAccount` representing both the contract and a potential beneficiary makes this test hard to read. Is there a more clear way we can name these?

Reply:
I refactored the tests so that benificiary and contract accounts are no longer shared.